pkg/seccomp/types.go

```diff
@@ -20,6 +20,18 @@ type Seccomp struct {
 	Flags         []string       `json:"flags,omitempty"`
 }
 
+const (
+	// SeccompFilterFlagLog is the filter to return actions except
+	// SECCOMP_RET_ALLOW should be logged. An administrator may override this
+	// filter flag by preventing specific actions from being logged via the
+	// /proc/sys/kernel/seccomp/actions_logged file. (since Linux 4.14)
+	SeccompFilterFlagLog = "SECCOMP_FILTER_FLAG_LOG"
```

Comment:
is there a risk that a malicious container could fill the log if this is enabled by default?

Reply:
The log message always has a limited line length accordingly to: https://github.com/torvalds/linux/blob/23d04328444a8fa0ca060c5e532220dac8e8bc26/kernel/auditsc.c#L2946-L2970

The kernel has an audit rate limit as well as backlog limit. Auditd has a file rotation in place as well. 

In theory users can still specify `SCMP_ACT_LOG` as default action, which would log all syscalls and not exclude the allowed ones.

Reply:
The log filter level may have a negative performance impact, I'll do some testing around this.

Reply:
### Benchmark results

#### Environment

- Linux 5.16.10
- cri-o 1.23.1
- Kubernetes 1.23.4

#### Test pod

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test-pod
spec:
  containers:
  - name: test-container
    securityContext:
      seccompProfile:
        type: RuntimeDefault
    image: nginx:1.21.6
```

#### Test

```
ab -n 100000 -c 100 http://$HOST/index.html
```

#### Results
| Seccomp Profile |  Test time (in sec) | Requests per second |
|-|-:|-:|
| RuntimeDefault | 5.0 | 19913.1 |
| RuntimeDefault (modified to log `sendfile` 1x per request) | 46.3 | 2156.4 |
| RuntimeDefault (modified to log `close` 2x per request) | 93.6 | 1068.7 |
| `{ "defaultAction": "SCMP_ACT_LOG" }` | 521.9 | 191.5 |
| Unconfined | 5.1 | 19323.4 |

It's interesting to see the impact of logging, which is executed in the kernel there:
https://github.com/torvalds/linux/blob/7e57714cd0ad2d5bb90e50b5096a0e671dec1ef3/kernel/auditsc.c#L2946-L2970

My local machine spikes audit CPU usage during the test, for example with `SCMP_ACT_LOG`:
![screenshot](https://user-images.githubusercontent.com/695473/155963843-0969657a-a16a-4dab-921d-3d020575607b.png)

Audit settings:
```
> sudo auditctl -s
enabled 1
failure 1
pid 803
rate_limit 0
backlog_limit 64
lost 163
backlog 65
backlog_wait_time 60000
loginuid_immutable 0 unlocked
```